From a73071b87a61acf1237da5ad9da59bae9316fb8c Mon Sep 17 00:00:00 2001
From: Chris Liddell <chris.liddell@artifex.com>
Date: Mon, 8 Oct 2018 09:24:22 +0100
Subject: [PATCH] Bug 699907: Reinstate GTK+ 2.x.x code.

The fix (from March 2018) for deprecated API calls in the GTK+ 3.x.x API
didn't account for the new, supported methods not being available in
GTK+ 2.x.x. Handle both cases, as we do elsewhere.
---
 psi/dxmain.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/psi/dxmain.c b/psi/dxmain.c
index 72c060e..250269a 100644
--- a/psi/dxmain.c
+++ b/psi/dxmain.c
@@ -354,17 +354,23 @@ static void window_resize(IMAGE *img)
 #endif
 
     if (!visible) {
-        guint width, height;
         /* We haven't yet shown the window, so set a default size
          * which is smaller than the desktop to allow room for
          * desktop toolbars, and if possible a little larger than
          * the image to allow room for the scroll bars.
          * We don't know the width of the scroll bars, so just guess. */
+#if !GTK_CHECK_VERSION(3, 0, 0)
+        gtk_window_set_default_size(GTK_WINDOW(img->window),
+            min(gdk_screen_width()-96, img->width+24),
+            min(gdk_screen_height()-96, img->height+24));
+#else
+        guint width, height;
         width = gtk_widget_get_allocated_width (img->window) - 96;
         height = gtk_widget_get_allocated_height (img->window) - 96;
         gtk_window_set_default_size(GTK_WINDOW(img->window),
             min(width, img->width+24),
             min(height, img->height+24));
+#endif
     }
 }
 
-- 
2.9.1
